### ISSUE TITLE
Rescue pool cannot reconnect in Redis 5

### ISSUE THREAD

**Issue by DennisWormald** [nevans/resque-pool#207]
When trying to update to redis 5 from redis 4.8, I get the following error: 

`"undefined method `reconnect' for #Redis::Client:0x000055c8e8ac9c78 (NoMethodError)"`

Which is originating from my lib/tasks/resque.rake file, with the following line as recommended in the readme of this project:

`Resque.redis._client.reconnect`

I believe that this is because reconnect is now a private method as of redis 5, as mentioned here: https://github.com/resque/resque/issues/1825

I hope we can resolve this, I wouldn't know how to fix it myself but would help if I could

**Comment by casperisfine** [comment]
@DennisWormald just replace that by `Resque.redis.close`. And you should be set.

**Comment by casperisfine** [comment]
> I also added ":inherit_socket => true" 

I very highly recommend not to do this, that will break your app in tons of ways.

**Comment by DennisWormald** [comment]
Big thanks to @casperisfine , this appears to work fine and I can leave everything else the same. Here is what I put in my rakefile (slightly refactored version of the code currently in readme.md, that is suggested to be put in "resque.rake" file). I left everything else the same, all working now as it used to I think, will test more over the next few days. I won't close the issue until we decide if the readme.md needs to be updated or not, as this is a redis-version dependent change.

```
task "resque:pool:setup" do
  # close any sockets or files in pool master
  ActiveRecord::Base.connection.disconnect!
  # and re-open them in the resque worker parent
  Resque::Pool.after_prefork do |job|
    ActiveRecord::Base.establish_connection
    Resque.redis.close
  end
end
```

**Comment by DennisWormald** [comment]
Ok I hadn't tested it properly, I tried it again this morning and I now have the following issue: https://github.com/resque/resque/issues/1826
And it seems a few other people are having the same problem. Will follow and see.

**Comment by DennisWormald** [comment]
okay all fixed with new updates to resque 2.4.0 and redis-rb 5.0.2 thanks @casperisfine and team
